src/core/Akka/Dispatch/Mailbox.cs

```diff
@@ -284,19 +301,22 @@ private void ProcessMailbox()
 
         private void ProcessMailbox(int left, long deadlineTicks)
         {
-            if (ShouldProcessMessage())
+            while (ShouldProcessMessage())
```

Comment:
Correct - I did this in a few instances where recursion showed up when I was performance profiling or threw a `StackOverflowException`. This was one of them. There were other instances of recursion within the mailbox (areas where we have some compare-and-swap operations) that occurred so infrequently that I left them as-is.